Bluetooth: mesh: transport: logic fix for unseg msgs

akredalen opened the pull request:
Fixed transport layer logic causing excessive processing of unsegmented messages in bt_mesh_trans_recv() calls.

**To do:**
- [x] Pass PTS tests (only known, unrelated issues failing)

PavelVPV commented on subsys/bluetooth/mesh/transport.c:
```diff
@@ -1666,6 +1666,16 @@ int bt_mesh_trans_recv(struct net_buf_simple *buf, struct bt_mesh_net_rx *rx)
+		/* Avoid further processing of unsegmented messages that are not a
+		 * local match nor a Friend match, with the exception of ctl messages.
+		 */
+		if (!rx->ctl) {
+			if (!rx->local_match && !rx->friend_match) {
+				return 0;
+			}
+		}
+
 		err = trans_unseg(buf, rx, &seq_auth);
```
I was too quick. It seems we should go further and call `net_buf_simple_restore`:
https://github.com/zephyrproject-rtos/zephyr/blob/40414f72e9fcec733b0ad4d046c3c042a579bb94/subsys/bluetooth/mesh/transport.c#L1677C2-L1677C24

Perhaps, something like this could work?
```suggestion
		if (!rx->ctl && !rx->local_match && !rx->friend_match) {
			err = -ENXIO;
		} else {
			err = trans_unseg(buf, rx, &seq_auth);
		}
```

alxelax replied:
Hm... Why is an error code returned for this case? Why is the error code not returned in case of segmented frame? Seems it makes behavior of this function for segmented and unsegmented frames different.
Why does it require buffer restoring if it is not proceeded further in the friend queue? 
Seems this change causes a lot of questions.

The changes that @akredalen suggested seem more logically correct. The only one issue on my point of view. It should be optimized to not check `local_match` and `friend_match` in both branches `if`.

PavelVPV replied:
Oh, yes, we don't change buf since we skip `trans_unseg`. Then it is fine!

The error code I suggested is was returned here https://github.com/zephyrproject-rtos/zephyr/blob/494f88070f6bb909389630264e4f89c71072ae53/subsys/bluetooth/mesh/transport.c#L799 and I guess this was used to bypass `bt_mesh_rpl_update` here: https://github.com/zephyrproject-rtos/zephyr/blob/494f88070f6bb909389630264e4f89c71072ae53/subsys/bluetooth/mesh/transport.c#L1058-L1060
Then yes, no need for error code.

akredalen committed:
Bluetooth: Mesh: Logic fix for recvd unseg msgs

Fixed transport layer logic causing excessive processing of
some unsegmented messages

Signed-off-by: Stine Åkredalen <stine.akredalen@nordicsemi.no>
